http/server.go

```
@@ -0,0 +1,76 @@
+package http
+
+import (
+	"net"
+	"net/http"
+	"net/url"
+
+	"github.com/benbjohnson/wtf"
+)
+
+// DefaultAddr is the default bind address.
+const DefaultAddr = ":3000"
+
+// Server represents an HTTP server.
+type Server struct {
+	ln net.Listener
+
+	// Handler to serve.
+	Handler *Handler
+
+	// Bind address to open.
+	Addr string
+}
+
+// NewServer returns a new instance of Server.
+func NewServer() *Server {
+	return &Server{
+		Addr: DefaultAddr,
+	}
+}
+
+// Open opens a socket and serves the HTTP server.
+func (s *Server) Open() error {
+	// Open socket.
+	ln, err := net.Listen("tcp", s.Addr)
+	if err != nil {
+		return err
+	}
+	s.ln = ln
+
+	// Start HTTP server.
+	go func() { http.Serve(s.ln, s.Handler) }()
+
+	return nil
+}
+
+// Close closes the socket.
+func (s *Server) Close() error {
+	if s.ln != nil {
+		s.ln.Close()
```

I find that other items get added to the `Close()` function and then if one errors and returns then it prevents the remaining items from being closed. It's not the case here but it's defensive programming for the future. But you're right, it should probably just return the error.